example/selection.js

```diff
@@ -415,8 +506,15 @@ const faceNormal = new THREE.Vector3();
 const toScreenSpaceMatrix = new THREE.Matrix4();
 const boxPoints = new Array( 8 ).fill().map( () => new THREE.Vector3() );
 const boxLines = new Array( 12 ).fill().map( () => new THREE.Line3() );
-const lassoSegments = [];
-const perBoundsSegments = [];
+
+/**
+ * Compute selected triangles:
+ *
+ * 1. Construct a list of screen space line segments that represent the lasso shape drawn by the user.
+ * 2. For every triangle in the geometry check if any part is within the lasso. If it is then consider the triangle selected.
+ *
+ * @see https://github.com/gkjohnson/three-mesh-bvh/issues/166#issuecomment-752194034
+ */
```

Comment:
I think it would be nice if this logic for detecting intersections in the BVH were pulled out into a separate file with something like the lasso shape being passed in as an argument so it's easier to follow.

Reply:
Yeah, indeed. Extracted it into a separate file